### PR TITLE
[6.0.2-devel] Add memkind support

### PR DESCRIPTION
**Pull request by szadam**
This change is [Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70)

### CODE REVIEW

File: src/rdb.c [hunk 1]
```diff
@@ -1514,10 +1521,8 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key) {
             double score;
             zskiplistNode *znode;
 
-            if ((sdsele = rdbGenericLoadStringObject(rdb,RDB_LOAD_SDS,NULL)) == NULL) {
-                decrRefCount(o);
```
Comment (PatKamin) [inline src/rdb.c:1518]: This shouldn't be removed. Missing adjustPmemThresholdCycle():
```
            if (zsetlen%100 == 0) {
                adjustPmemThresholdCycle();
            }

            if ((sdsele = rdbGenericLoadStringObject(rdb,RDB_LOAD_SDS,NULL)) == NULL) {
                decrRefCount(o);
                return NULL;
            }
```

Comment (szadam) [reply src/rdb.c:1518]: done

**Review by PatKamin — CHANGES_REQUESTED** [review]
> Reviewed 1 of 11 files at r11.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70)* status: 1 of 33 files reviewed, 3 unresolved discussions (waiting on @PatKamin and @szadam)

___
*[`src/rdb.c` line 1523 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJsiLzEQkLzAq8oKtw:-NAJsiLzEQkLzAq8oKtx:bf20ufi) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1523)):*
> ```C
>             double score;
>             zskiplistNode *znode;
>             
> ```

Remove tabulation.
___
*[`src/rdb.c` line 1532 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJsk2n8wIen_LSLafP:-NAJsk2n8wIen_LSLafQ:b3kqx) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1532)):*
> ```C
>                 return NULL;
>             }
>             
> ```

.
___
*[`src/rdb.c` line 1533 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJtNjK6viK9EclY8RT:-NAJtNjK6viK9EclY8RU:b3sucgw) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1533)):*
> ```C
>             }
>             
>             if ((sdsele = rdbGenericLoadStringObject(rdb,RDB_LOAD_SDS,NULL))
> ```

This is not a part of this commit. Remove.



_Code quote:_
```C
            if ((sdsele = rdbGenericLoadStringObject(rdb,RDB_LOAD_SDS,NULL))
                == NULL) return NULL;
```

**Review by szadam — COMMENTED** [review]
> Reviewed 1 of 11 files at r21, 1 of 1 files at r22, 2 of 3 files at r23, 2 of 2 files at r24, 1 of 2 files at r25, 1 of 1 files at r27, 1 of 1 files at r28, 2 of 2 files at r30, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70)* status: 11 of 33 files reviewed, 3 unresolved discussions (waiting on @PatKamin)

___
*[`src/rdb.c` line 1523 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJsiLzEQkLzAq8oKtw:-NAcdz0f3E_kgF0Vzyql:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1523)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Remove tabulation.
</blockquote></details>

Done.
___
*[`src/rdb.c` line 1532 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJsk2n8wIen_LSLafP:-NAcdzKr-CZ0kAMS-yOv:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1532)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.
___
*[`src/rdb.c` line 1533 at r20](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-NAJtNjK6viK9EclY8RT:-NAce1vKDu2yzuO05Sq8:b-896fix) ([raw file](https://github.com/tieredmemdb/tieredmemdb/blob/460cbc0d5e83d256394eddddd726a18aae8284dd/src/rdb.c#L1533)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

This is not a part of this commit. Remove.
</blockquote></details>

Done.

**Review by PatKamin — APPROVED** [review]
:lgtm:

> Reviewed 1 of 30 files at r1, 1 of 11 files at r21.
> *[Reviewable](https://reviewable.io/reviews/tieredmemdb/tieredmemdb/70#-:-NAmYH5t873xIcf2_vLK:bnfp4nl)* status: 12 of 33 files reviewed, all discussions resolved (waiting on @PatKamin)

**Pull request merged**